### DISCUSSION TITLE
The sequence in the data

### DISCUSSION THREAD

**Discussion by DAV95s** [pocketbase/pocketbase#4922] [Q&A] [answered]
Does pocketbase guarantee the sequence of images in multifilefield?
Why is the field order not preserved in json objects?

**Comment by ganigeorgiev** [comment]
> Does pocketbase guarantee the sequence of images in multifilefield?

Yes, arrayble fields (`file`, `select`, `relation`) should preserve the order of the values they hold.

> Why is the field order not preserved in json objects?

As mentioned above - it should be, if not provide reproducible steps.

Please don't treat the discussions as LLM chat and try to elaborate as much as possible to avoid the needless back-and-forth and to understand the broader picture of the issue.

**Reply by DAV95s** [reply]
I'm sorry, it's hard for me to express my thoughts well(.
I saved this entry yesterday
{
  "3 days": 4300,
  "7 days": 5900,
  "14 days": 7900,
  "30 days": 9900
}
But today, it's look:
{
  "14 days": 7900,
  "3 days": 4300,
  "30 days": 9900,
  "7 days": 5900
}

**Reply by ganigeorgiev** [reply, marked as answer]
On second read, if you meant the field props in the returned record's json - then No, but this shouldn't matter because maps are unordered per spec, aka:

```js
{
    "id": "123",
    "title": "456"
}
// is considered the same as
{
    "title": "456",
    "id": "123"
}
```
